packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -134,12 +134,11 @@ interface PendingBlob {
 	handleP: Deferred<IFluidHandle<ArrayBufferLike>>;
 	uploadP: Promise<ICreateBlobResponse>;
 	localUploadTime?: number;
-	serverUploadTime?: number;
 	minTTLInSeconds?: number;
 }
 
 export interface IPendingBlobs {
-	[id: string]: { blob: string };
+	[id: string]: { blob: string; localUploadTime?: number; minTTLInSeconds?: number };
```

Comment:
is there an advantage to keep these separate over just having like expiry time (localTime + minTTL)?

Reply:
now that i think about it a bit more, i wonder if we want to take a more conservative approach, and only track this data in-memory. So if we go offline, then come back online with the browser still open we can use TTL, but if  we close the container we always re-upload.

Reply:
In fact, if we always reupload I don't see any scenario in which TTL is needed. If we optimize and only reupload in case the TTL has expired (or about to), then check for expiration before the BlobAttach op makes a little bit more sense.